src/ui/WindowViewModelBase.hh

```diff
@@ -0,0 +1,23 @@
+#ifndef RA_UI_WINDOW_VIEW_MODEL_BASE_H
+#define RA_UI_WINDOW_VIEW_MODEL_BASE_H
+#pragma once
+
+namespace ra {
+namespace ui {
+
+typedef struct Position
+{
+    int X;
+    int Y;
+} Position;
```

Comment:
No, it's just the `typedef struct ` part that looked strange. You can direct initialize a struct in C++ like with yours you can put (example).

```
Position xy1{-32, 432}; // you can change this how ever you like
constexpr Position xy1{-32, 432}; // can't change this one, it's good if you don't want it to change
```

 As long as you don't define any constructors you do the above (all the defaults should be `constexpr `anyway). Also constexpr doesn't force an instance to be constant, but it can be. You don't need getters and setters with the first comment in the post.

It's honestly not too a big a deal, I'm just not sure if `struct` or `typedef struct` has the same effect

Reply:
removed typedef, added initializers.